stl/inc/atomic

```diff
@@ -714,40 +784,40 @@ struct _Atomic_storage<_Ty, 8> { // lock-free using 8-byte intrinsics
         _As_bytes = __iso_volatile_load64(_Mem);
         _Compiler_or_memory_barrier();
 #endif
-        return reinterpret_cast<_Ty&>(_As_bytes);
+        return reinterpret_cast<_TVal&>(_As_bytes);
     }
 
-    _NODISCARD _Ty load(const memory_order _Order) const noexcept { // load with given memory order
+    _NODISCARD _TVal load(const memory_order _Order) const noexcept { // load with given memory order
         const auto _Mem = _Atomic_address_as<long long>(_Storage);
 #ifdef _M_ARM
         long long _As_bytes = __ldrexd(_Mem);
 #else
         long long _As_bytes = __iso_volatile_load64(_Mem);
 #endif
         _Load_barrier(_Order);
-        return reinterpret_cast<_Ty&>(_As_bytes);
+        return reinterpret_cast<_TVal&>(_As_bytes);
     }
 
 #if defined(_M_IX86) && defined(__clang__) // TRANSITION, LLVM-46595
-    _Ty exchange(const _Ty _Value, const memory_order _Order = memory_order_seq_cst) noexcept {
+    TVal exchange(const _TVal _Value, const memory_order _Order = memory_order_seq_cst) noexcept {

```

Comment:
There is a bug here - the first `TVal` is missing a leading underscore.

I am not sure what's happening with the `complex.special` tests, which appear to be totally unrelated, and have several confusing negations of "libc++ expects this to fail to compile" "but our implementation improperly allows it" "so we expect this to FAIL by successfully compiling" "but it's XPASSing by failing to compile" (I think that's it, anyways).

Reply:
Yeah, good catch. Merge error.

Reply:
And thanks for interpreting this triple negation, now I see.